src/monetization.js

```diff
@@ -98,6 +102,11 @@ export class Monetization extends EventEmitter {
 
         const sortedIds = productIds.sort();
         const cacheKey = this._accessCacheKey(productIds, userId);
+        const referrer = this.window && this.window.location && this.window.location.referer;
+        if (isReferredFromAccountPages(referrer, this.env) && !this._clearedCaches[cacheKey]) {
+            this.clearCachedAccessResult(productIds, userId);
+            this._clearedCaches[cacheKey] = true;
+        }
```

Comment:
How about doing this in the constructor too, but iterate over the cache and find keys that match? Granted, it won't be perfect _(given that we'd need to prune any keys that look like `/prd_[a-z0-9-]+(?:,[a-z0-9-]+)_[a-z0-9]+/i`)_. But it would ensure we don't have any stray cache entries staying around

Reply:
@PAkerstrand I agree that would be a better solution, but that would require changing the `Cache` interface a bit.
I think the cache instances should have a `clear` method.
I wanted to keep the changes in this PR small, to get get the discussion going first.